### PR TITLE
enable sirius

### PR DESCRIPTION
**Pull request by ltalirz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ltalirz** [comment]
@conda-forge-admin, please rerender

**Comment by ltalirz** [comment]
@adhocman cp2k also has a serial (openmp-enabled) build.
if we want the serial cp2k version also to be linked against sirius, I guess we should have a serial sirius version.
what do you think?

for the moment we can add sirius only to the parallel cp2k build

**Comment by ltalirz** [comment]
In any case, we'll still need to wait until the cp2k 8.2 release since cp2k 8.1 only supports libxc 4.3.x

**Comment by AdhocMan** [comment]
> @AdhocMan cp2k also has a serial (openmp-enabled) build.
> if we want the serial cp2k version also to be linked against sirius, I guess we should have a serial sirius version.
> what do you think?
> 
> for the moment we can add sirius only to the parallel cp2k build

Sirius always requires MPI, so I think it's best to leave it disabled for the serial version.

**Comment by ltalirz** [comment]
@AdhocMan Could you please let me know how to properly link against sirius?

I've tried added the sirius includes and am linking against the library but it [complains about undefined symbols](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=329984&view=logs&j=bc0b25a1-0ea6-5be3-17ad-1212c4b469de&t=578628e4-10f7-5e57-f7f0-bcdd728dd32a&l=2327)

**Comment by AdhocMan** [comment]
It looks like the changes required for CP2K to use the latest SIRIUS version did not make into the 8.2 release. Since conda does not seem to support multiple versions very well, I'd suggest applying a patch to CP2K. Here is a patch file that should work: https://gist.github.com/AdhocMan/bdf3f1065af4b5ff7aad0b00f875179f

**Comment by ltalirz** [comment]
Thanks for that patch!

> Since conda does not seem to support multiple versions very well,

Just to understand, you are saying this could be handled on the conda side? What feature is missing here?

**Comment by AdhocMan** [comment]
> Just to understand, you are saying this could be handled on the conda side? What feature is missing here?

An older version of SIRIUS should work. I just had the impression, that conda usually only has one version of a package. Looking into it now, it can be solved by creating an additional branch in the sirius feedstock for the required version.

**Comment by AdhocMan** [comment]
I'm not sure why the patch isn't working, maybe @mtaillefumier could help here. Alternatively, we could try it with SIRIUS version 7.0.2.

**Comment by mtaillefumier** [comment]
I think the patch might be faulty.

**Comment by mtaillefumier** [comment]
the previous patch was referring to something related to dbcsr, so I had to manually modify the patch. This new patch works on the latest master but should work on V8.2 as well

[sirius.patch.txt](https://github.com/conda-forge/cp2k-feedstock/files/6570999/sirius.patch.txt)

**Comment by mtaillefumier** [comment]
that's really strange since the patch removes exactly these two functions calls. I wonder if the pipeline apply the patch or not.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ source:
   sha256: 2e24768720efed1a5a4a58e83e2aca502cd8b95544c21695eb0de71ed652f20a
   patches:
     - macosxtest.patch  # [osx]
+    # remove in release following cp2k 8.2
+    - sirius.patch  # [osx]
```
Comment (AdhocMan) [inline recipe/meta.yaml]:
```suggestion
    - sirius.patch
```
The patch is only applied on macOS right now.

**Comment by ltalirz** [comment]
Oops... thanks for pointing out this oversight ;-)

**Comment by ltalirz** [comment]
[psmp build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=331210&view=logs&j=bc0b25a1-0ea6-5be3-17ad-1212c4b469de&t=578628e4-10f7-5e57-f7f0-bcdd728dd32a&l=6792)
```
Number of FAILED  tests 1
Number of WRONG   tests 4
Number of CORRECT tests 2963
Total number of   tests 2968
GREPME 1 4 2963 0 2968 X
```
It seems that 4 of the sirius tests yield results outside the tolerance (some of them by more than an order of magnitude).

One of the tests failed:
```
$SRC_DIR/regtesting/Linux-x86-64-conda/psmp/TEST-Linux-x86-64-conda-psmp-2021-06-01_11-52-50/SIRIUS/regtest-1/He-full-potential.inp.out
+------------------------------+

Linearization energies
Atom : He, class id : 0
augmented waves
n =  1   l =  0   order = 0   enu =    -1.564330  +
n =  1   l =  0   order = 1   enu =    -1.564330  +
local orbitals
n =  1   l =  0   order = 0   enu =    -1.564330  +
n =  1   l =  0   order = 1   enu =    -1.564330  +

[get_singular_components] smallest eigen-value of the singular components:   0.0012990509462977
[diag_full_potential_first_variation_davidson] iterative solver tolerance:     0.000050000000
[diag_full_potential_first_variation_davidson] step: 0, current subspace size: 61, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 1, current subspace size: 101, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 2, current subspace size: 141, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 3, current subspace size: 181, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 4, current subspace size: 221, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 5, current subspace size: 261, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 6, current subspace size: 301, maximum subspace size: 341
[diag_full_potential_first_variation_davidson] step: 7, current subspace size: 341, maximum subspace size: 341
[solve] Lowest band energies
ik :  0,   -45.610114  -45.607113  -45.600415  -45.564806  -44.589979  -44.565682  -44.491899  -43.994386  -43.967995  -43.879579
terminate called after throwing an instance of 'std::runtime_error'
terminate called after throwing an instance of 'std::runtime_error'
  what():    what():  
=== Fatal error at line 243 of file /home/conda/feedstock_root/build_artifacts/sirius_1621525035642/work/src/k_point/k_point_set.cpp ===
search of band occupancies failed after 10000 steps


=== Fatal error at line 243 of file /home/conda/feedstock_root/build_artifacts/sirius_1621525035642/work/src/k_point/k_point_set.cpp ===
search of band occupancies failed after 10000 steps
```

Any comments @AdhocMan ?

**Comment by mtaillefumier** [comment]
[sirius.patch.txt](https://github.com/conda-forge/cp2k-feedstock/files/6582703/sirius.patch.txt)
this should fix the problem hopefully.

**Comment by mtaillefumier** [comment]
You can stop the pipeline if possible because it will fail. the Fe-DOS.inp has a wrong value for one of the parameters (I deleted it then added it again in my patch forgetting about changing the value of the parameter). 
[sirius.patch.txt](https://github.com/conda-forge/cp2k-feedstock/files/6583070/sirius.patch.txt)

**Comment by ltalirz** [comment]
Thanks! I also invited you as a collaborator to my fork - in case you notice any further issues, feel free to push directly to the branch.

**Comment by mtaillefumier** [comment]
I will need to modify the patch again later but I need to test the changes before. But this should not be blocking you. If you do not mind I prefer to put the patch in the comments section to avoid screwing up your branch. And it is better for review as well

**Comment by ltalirz** [comment]
thanks, indeed [all the tests pass now](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=331787&view=logs&j=bc0b25a1-0ea6-5be3-17ad-1212c4b469de&t=578628e4-10f7-5e57-f7f0-bcdd728dd32a&l=6669)

```
Number of FAILED  tests 0
Number of WRONG   tests 0
Number of CORRECT tests 2968
Total number of   tests 2968
GREPME 0 0 2968 0 2968 X
```

@mtaillefumier Should this be merged?

**Comment by mtaillefumier** [comment]
Go ahead. It needs more work of SIRIUS side and it is not a limiting factor either.

**Pull request merged**